contracts/token/ERC1155/ERC1155TokenReceiver.sol

```diff
@@ -0,0 +1,13 @@
+pragma solidity ^0.5.0;
+
+import "./IERC1155TokenReceiver.sol";
+import "../../introspection/ERC165.sol";
+
+contract ERC1155TokenReceiver is ERC165, IERC1155TokenReceiver {
```

Comment:
This, I've chosen to resolve in a different manner than prescribed.

First, I've renamed `[I]ERC1155TokenReceiver` to `[I]ERC1155Receiver` in order to keep these contracts consistent with the naming of `ERC721Receiver` in OZ. (The original EIPs for ERC721 and ERC1155 both name the receiver `ERC*TokenReceiver`, so I did the rename to keep the contracts consistent internally in the repo.)

Then, I've created _another_ contract called `ERC1155Holder`. The reasoning for having three different receivers was as follows:

* `IERC1155Receiver` is a pure interface. Since the EIP specifies that the interface must also be EIP-165 compatible, I've extended the interface from there. The main use for the interface is to ergonomically handle calling checks on contract accounts which are potentially 1155 receivers.
* `ERC1155Receiver` extends from the pure interface and describes part of its construction. Specifically, it ensures that the contract's implementation of EIP-165 is correct as far as the 1155 spec goes. It does not, however, specify the transfer acceptance/rejection policy, which is why it's abstract.
* `ERC1155Holder` was created to mirror the existing `ERC721Holder` implementation, which automatically accepts all transfers.